### PR TITLE
fix(argo-rollouts): avoid label exceeding maximum length

### PR DESCRIPTION
**Pull request by steenzout**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-rollouts/templates/_helpers.tpl [hunk 1]
```diff
@@ -31,14 +31,28 @@ Create chart name and version as used by the chart label.
...
+
 {{/*
 Common labels
 */}}
 {{- define "argo-rollouts.labels" -}}
 helm.sh/chart: {{ include "argo-rollouts.chart" . }}
 {{ include "argo-rollouts.selectorLabels" . }}
 {{- if .Chart.AppVersion }}
-app.kubernetes.io/version: {{ default .Chart.AppVersion $.Values.controller.image.tag | quote }}
+app.kubernetes.io/version: {{ include "argo-rollouts.chart_version_label" . }}
```
Comment (steenzout) [inline charts/argo-rollouts/templates/_helpers.tpl:55]: does it make sense to only set this label when `.Chart.AppVersion` is set?

if I override the `image.tag` instead? it shouldn't be set? or should it be set to the `image.tag` value (without SHA digest)?

File: charts/argo-rollouts/templates/_helpers.tpl [hunk 2]
```diff
@@ -31,14 +31,28 @@ Create chart name and version as used by the chart label.
...
+{{- regexReplaceAll "[^a-zA-Z0-9-_.]+" (regexReplaceAll "@sha256:[a-f0-9]+" (default .Chart.AppVersion $.Values.controller.image.tag) "") "" | trunc 63 | quote -}}
+{{- end -}}
+
 {{/*
 Common labels
 */}}
 {{- define "argo-rollouts.labels" -}}
 helm.sh/chart: {{ include "argo-rollouts.chart" . }}
 {{ include "argo-rollouts.selectorLabels" . }}
 {{- if .Chart.AppVersion }}
```
[the remark this answers is not in the document]
Comment (steenzout) [reply charts/argo-rollouts/templates/_helpers.tpl:54]: `argo-workflows` and `argo-events` actually always set this label. shouldn't this chart do the same?

[the remark this answers is not in the document]
Comment (jmeridth) [reply charts/argo-rollouts/templates/_helpers.tpl:54]: Consistency is good. I say add it.  @mkilchhofer do you concur?

[the remark this answers is not in the document]
Comment (mkilchhofer) [reply charts/argo-rollouts/templates/_helpers.tpl:54]: Yes, I also like consistency. If you have time to adapt it @steenzout I would appreciate it 🙏

**Review by mkilchhofer — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-rollouts/Chart.yaml [hunk 3]
```diff
@@ -15,4 +15,5 @@ maintainers:
     url: https://argoproj.github.io/
 annotations:
   artifacthub.io/changes: |
-    - "[Added]: Add support for topologySpreadConstraints"
+    - "[Fixed]: avoid app.kubernetes.io/version kubernetes label from exceeding maximum length (63)
```
Comment (mkilchhofer) [inline charts/argo-rollouts/Chart.yaml:18]:
```suggestion
    - "[Fixed]: avoid app.kubernetes.io/version kubernetes label from exceeding maximum length (63)"
```

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**